src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -535,7 +535,7 @@ private IMemberReferenceExpression CreateBoundObjectInitializerMemberOperation(B
             Lazy<IOperation> instance = new Lazy<IOperation>(() => new InstanceReferenceExpression(
                 semanticModel: _semanticModel,
                 syntax: boundObjectInitializerMember.Syntax,
-                type: boundObjectInitializerMember.MemberSymbol.ContainingType,
+                type: boundObjectInitializerMember.InstanceSymbol,
```

Comment:
Would it be sufficient to walk up to the containing `BoundObjectInitializerExpression` and use the `Type` of that node instead?

Reply:
How though? We don't have a parent operation here to use, and doing that via syntax feels like it would be very fragile to me.

Reply:
I see. In that case, holding on to the receiver type makes sense.